packages/compass-saved-aggregations-queries/src/components/aggregations-queries-list.tsx

```diff
@@ -4,20 +4,22 @@ import type { ConnectedProps } from 'react-redux';
 import type { ThunkDispatch } from 'redux-thunk';
 import { VirtualGrid, H2, css, spacing } from '@mongodb-js/compass-components';
 import { fetchItems } from '../stores/aggregations-queries-items';
+import type { Item } from '../stores/aggregations-queries-items';
 import { openSavedItem } from '../stores/open-item';
 import type { RootActions, RootState } from '../stores/index';
 import { SavedItemCard, CARD_WIDTH, CARD_HEIGHT } from './saved-item-card';
 import type { SavedItemCardProps, Action } from './saved-item-card';
 import OpenItemModal from './open-item-modal';
+import { useGridHeader } from '../hooks/use-grid-header';
 
 const ConnectedItemCard = connect<
   Omit<SavedItemCardProps, 'onAction'>,
   Pick<SavedItemCardProps, 'onAction'>,
-  { index: number },
+  { index: number; items: Item[] },
   RootState
 >(
-  ({ savedItems: { items } }, props) => {
-    const item = items[props.index];
+  (_state, { index, items }) => {
```

Comment:
If we are not using state at all, why do we need mapState here? Ideally we would somehow keep the pattern here instead of passing all the items to the component